src/main/java/thePackmaster/actions/pixiepack/DrawSpecificCardAction.java

```diff
@@ -0,0 +1,32 @@
+package thePackmaster.actions.pixiepack;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.characters.AbstractPlayer;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+
+public class DrawSpecificCardAction extends AbstractGameAction {
+    AbstractCard toDraw;
+    AbstractPlayer player;
+    public DrawSpecificCardAction(AbstractCard C)
+    {
+        this.actionType = ActionType.CARD_MANIPULATION;// 21
+        this.duration = this.startDuration = Settings.ACTION_DUR_FAST;
+        this.player = AbstractDungeon.player;
+        toDraw = C;
+    }
+    @Override
+    public void update() {
+        if (!this.player.drawPile.isEmpty() || !this.player.hand.contains(toDraw))
+        {
+            if (this.player.hand.size() == 10) {// 43
```

Comment:
There are effects that can modify a player's hand size, so this should check against `BaseMod.MAX_HAND_SIZE`.

Reply:
Fixed